ddl/ddl_api.go

```diff
@@ -4373,10 +4373,11 @@ func (d *ddl) AlterTablePartitioning(ctx sessionctx.Context, ident ast.Ident, sp
 		BinlogInfo: &model.HistoryInfo{},
 		Args:       []interface{}{partNames, newPartInfo},
 		ReorgMeta: &model.DDLReorgMeta{
-			SQLMode:       ctx.GetSessionVars().SQLMode,
-			Warnings:      make(map[errors.ErrorID]*terror.Error),
-			WarningsCount: make(map[errors.ErrorID]int64),
-			Location:      &model.TimeZoneLocation{Name: tzName, Offset: tzOffset},
+			SQLMode:           ctx.GetSessionVars().SQLMode,
+			Warnings:          make(map[errors.ErrorID]*terror.Error),
+			WarningsCount:     make(map[errors.ErrorID]int64),
+			Location:          &model.TimeZoneLocation{Name: tzName, Offset: tzOffset},
+			ResourceGroupName: ctx.GetSessionVars().ResourceGroupName,
```

Comment:
Do we need to add it to `MultiSchemaChange`'s `ReorgMeta`?

Reply:
Besides, could we add a `NewReorgMeta` to unify the assignment? This helps reduce the possibility of missing parameter assignments later on

Reply:
> Besides, could we add a NewReorgMeta to unify the assignment? This helps reduce the possibility of missing parameter assignments later on

lgtm

Reply:
fixed all in https://github.com/pingcap/tidb/pull/46756/commits/d3ad2bf7bb2f8ebb47be4fc5bea197dfa2307695